src/server/replica.cc

```diff
@@ -1195,23 +1191,7 @@ auto DflyShardReplica::TransactionReader::NextTxData(JournalReader* reader, Cont
       return std::nullopt;
     }
 
-    // Check if journal command can be executed right away.
-    // Expiration checks lock on master, so it never conflicts with running multi transactions.
-    if (res->opcode == journal::Op::EXPIRED || res->opcode == journal::Op::COMMAND ||
-        res->opcode == journal::Op::PING)
-      return TransactionData::FromSingle(std::move(res.value()));
-
-    // Otherwise, continue building multi command.
-    DCHECK(res->opcode == journal::Op::MULTI_COMMAND || res->opcode == journal::Op::EXEC);
-    DCHECK(res->txid > 0);
-
-    auto txid = res->txid;
-    auto& txdata = current_[txid];
-    if (txdata.AddEntry(std::move(res.value()))) {
-      auto out = std::move(txdata);
-      current_.erase(txid);
-      return out;
-    }
+    return TransactionData::FromSingle(std::move(res.value()));
```

Comment:
what was removed? (I am not familiar with this code)

Reply:
we dont accumulate the multi commands untill we get the exec opcode. instead we just execute them when we get them

Reply:
all the code above for collecting journal entries because we don't collect them anymore